app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt

```diff
@@ -153,9 +162,44 @@ class BrowserToolbarIntegration(
         toolbar.display.setUrlBackground(urlBackground)
     }
 
+    private fun setActionButtons(context: Context) {
+        if (store.state.findCustomTabOrSelectedTab(customTabId)?.isCustomTab() == false) {
+            val eraseAction = BrowserToolbar.Button(
+                imageDrawable = AppCompatResources.getDrawable(
+                    context,
+                    R.drawable.mozac_ic_delete
+                )!!,
+                contentDescription = context.getString(R.string.content_description_erase),
+                iconTintColorResource = R.color.primaryText,
+                listener = {
+                    val openedTabs = store.state.tabs.size
+                    TabCount.eraseButtonTapped.record(TabCount.EraseButtonTappedExtra(openedTabs))
+                    eraseActionListener.invoke()
+                }
+            )
+            toolbar.addNavigationAction(eraseAction)
+
+            store.flowScoped { flow ->
+                flow.ifChanged { state -> state.tabs.size }
+                    .collect {
+                        if (store.state.tabs.size > 1) {
+                            val tabsAction = TabCounterToolbarButton(
+                                lifecycleOwner = fragment,
+                                showTabs = {
+                                    toolbar.hideKeyboard()
+                                    tabCounterListener.invoke()
+                                },
+                                store = store
+                            )
+                            toolbar.addBrowserAction(tabsAction)
+                        }
```

Comment:
I also wonder if we need to remove the button too. In the past when navigating back in a tab ... if we couldn't navigate back anymore then we closed that tab. But that doesn't seem to be the case anymore, now the whole session gets removed. That may be a bug? 🤔

Reply:
yep, seems a regression added by me in this pr. Thanks